src/services/withParams.ts

```diff
@@ -15,9 +15,16 @@ type WithParamsParamsOutput<
   TValue extends string,
   TParams extends Record<string, Param | undefined> = Record<never, never>
 > = TValue extends `${string}${ParamStart}${infer TParam}${ParamEnd}${infer Rest}`
-  ? Record<TParam, ExtractWithParamsParamType<TParam, TParams>> & WithParamsParamsOutput<Rest, TParams>
+  ? Record<ExtractParamName<TParam>, ParamTypeOrDefault<ExtractParamName<TParam>, TParams>> & WithParamsParamsOutput<Rest, TParams>
   : {}
 
+type ParamTypeOrDefault<
+  TKey extends string,
+  TParams extends Record<string, Param | undefined>
+> = TKey extends keyof TParams
+  ? TParams[TKey]
+  : StringConstructor
```

Comment:
What's the thought behind defaulting the params to the StringConstructor here? I believe we have other types that fallback to `string` for param types if a param is missing. But now it'll always be defined right?

Reply:
however, this IS accurate. Once `withParams` is called, any params not defined WILL be defined as `String`

Reply:
yep that makes sense, and https://github.com/kitbagjs/router/pull/464/commits/c153bd4890241d0f969af111fbdafcb579b47c3b simplifies this more how I'd expect. 

We probably have downstream types that are still expecting a param to possibly not be there that we can maybe simplify since we're normalizing `WithParams` to always include the param.